packages/eslint-plugin/src/rules/no-explicit-any.ts

```diff
@@ -170,36 +176,88 @@ export default createRule<Options, MessageIds>({
       );
     }
 
+    /**
+     * Checks if the node is within a keyof any expression
+     * @param node the node to be validated.
+     * @returns true if the node is within a keyof any expression, false otherwise
+     * @private
+     */
+    function isNodeWithinKeyofAny(node: TSESTree.Node): boolean {
+      return (
+        node.parent?.type === AST_NODE_TYPES.TSTypeOperator &&
+        node.parent.operator === 'keyof'
+      );
+    }
+
+    /**
+     * Creates a fixer that replaces a keyof any with PropertyKey
+     * @param node the node to be fixed.
+     * @returns a function that will fix the node.
+     * @private
+     */
+    function createPropertyKeyFixer(node: TSESTree.Node) {
```

Comment:
I should have mentioned this more explicitly, sorry: by _"refactor the code"_ I meant refactoring either the runtime logic _or_ the types. In this case `createPropertyKeyFixer` is only ever called with a node that we know to have a `.parent`. So we can do this:

```suggestion
    function createPropertyKeyFixer(node: MakeRequired<TSESTree.Node, 'parent'>) {
```

Or, even better, we know `createPropertyKeyFixer` is only ever called with a `TSESTree.TSAnyKeyword`. So we can do this instead:

```suggestion
    function createPropertyKeyFixer(node: TSESTree.AnyKeyword) {
```

...and either way, the `as TSESTree.TSTypeOperator` is unnecessary.

Change request: switch to `: TSESTree.AnyKeyword` and remove the `as`.